### PR TITLE
Upgrade kubernetes-client-api to 6.8.1

### PR DESCRIPTION
**Pull request by julieheard**
Downstream of https://github.com/jenkinsci/kubernetes-client-api-plugin/pull/221

### Testing done
I ran mvn verify to test locally and all working, all tests passing

```[tasklist]
### Submitter checklist
- [x] Make sure you are opening from a **topic/feature/bugfix branch** (right side) and not your main branch!
- [x] Ensure that the pull request title represents the desired changelog entry
- [x] Please describe what you did
- [ ] Link to relevant issues in GitHub or Jira
- [x] Link to relevant pull requests, esp. upstream and downstream changes
- [ ] Ensure you have provided tests - that demonstrates feature works or fixes the issue
```

### CODE REVIEW

File: pom.xml [hunk 1]
```diff
@@ -234,7 +234,11 @@
         </exclusion>
       </exclusions>
     </dependency>
-
+    <dependency>
+      <groupId>io.jenkins.plugins</groupId>
+      <artifactId>snakeyaml-api</artifactId>
+      <version>2.0-999999-SNAPSHOT</version>
```
Comment (rsandell) [inline pom.xml:240]: An incremental version was deployed https://ci.jenkins.io/job/Plugins/job/snakeyaml-api-plugin/job/PR-94/1/pipeline-console/ but how to figure out what that version is I don't know.

Comment (rsandell) [reply pom.xml:240]: Found it!
```suggestion
      <version>2.1-110.v4960f4074978</version>    
```

File: pom.xml [hunk 2]
```diff
@@ -141,7 +141,7 @@
     <dependency>
       <groupId>org.jenkins-ci.plugins</groupId>
       <artifactId>kubernetes-client-api</artifactId>
-      <version>6.3.1-206.v76d3b_6b_14db_b</version>
+      <version>6.8.1-220.v1f66736f9a_42</version>
```
Comment (Vlatombe) [inline pom.xml:144]:
```suggestion
      <version>6.8.1-224.vd388fca_4db_3b_</version>
```

**Comment by Vlatombe** [comment]
@julieheard Need to bump `jenkins.version` here.
```
[2023-08-29T13:30:25.903Z] Caused by: org.apache.maven.plugin.MojoExecutionException: Dependency org.jenkins-ci.plugins:kubernetes-client-api:jar:6.8.1-224.vd388fca_4db_3b_ requires Jenkins 2.401.1 or higher.
```

**Comment by Vlatombe** [comment]
@julieheard I can't update this PR so you need to fix it.

**Comment by julieheard** [comment]
I am on it, when testing locally it also wants bumps for bouncycastle-api:2.27, okhttp-api:4.10.0-132.v7a_7b_91cef39c and jackson2-api:2.15.1-344.v6eb_55303dc3e

**Comment by julieheard** [comment]
Ok to fix this I had to:

**Comment by julieheard** [comment]
Due to the Jenkins bump I have also bumped Kotlin and included three new dependencies (bouncycastle-api, jackson2-api, okhttp-api) as they were creating RequireUpperBoundDeps errors.

I came accross a strange issue with a test, `mvn verify` it was failing intermittently with this message:
```
[ERROR] Tests run: 10, Failures: 0, Errors: 2, Skipped: 0, Time elapsed: 2.661 s <<< FAILURE! -- in com.cloudbees.jenkins.plugins.kubernetes_credentials_provider.convertors.CertificateCredentialsConvertorTest
[ERROR] com.cloudbees.jenkins.plugins.kubernetes_credentials_provider.convertors.CertificateCredentialsConvertorTest.failsToConvertWhenCertificateCorruptBase64 -- Time elapsed: 0.015 s <<< ERROR!
java.lang.NoSuchMethodError: 'org.hamcrest.Matcher org.hamcrest.CoreMatchers.allOf(org.hamcrest.Matcher, org.hamcrest.Matcher)'
[ERROR] com.cloudbees.jenkins.plugins.kubernetes_credentials_provider.convertors.CertificateCredentialsConvertorTest.failsToConvertWhenCertificateCorruptPKCS12 -- Time elapsed: 0.006 s <<< ERROR!
java.lang.NoSuchMethodError: 'org.hamcrest.Matcher org.hamcrest.CoreMatchers.allOf(org.hamcrest.Matcher, org.hamcrest.Matcher)'
```
Digging into this I think it is a conflict between junit and hamcrest, I do not think it is due to the changes made in this PR.

**Comment by Vlatombe** [comment]
```
java.lang.NoSuchMethodError: 'org.hamcrest.Matcher org.hamcrest.CoreMatchers.allOf(org.hamcrest.Matcher, org.hamcrest.Matcher)'
```

Can't reproduce on my side, and I only see one hamcrest copy in the classpath (through junit)

```
+- junit:junit:jar:4.13.2:test
|  \- org.hamcrest:hamcrest-core:jar:2.2:test
+- org.hamcrest:hamcrest:jar:2.2:test
```

**Comment by julieheard** [comment]
I have re-ran this test quite a few times today and also now can't reproduce. Must have been a small glitch in the classpath or something silly. Hopefully this PR is now ready to merge 🙂

File: pom.xml [hunk 3]
```diff
@@ -234,14 +234,29 @@
...
+    <dependency>
+      <groupId>org.jenkins-ci.plugins</groupId>
+      <artifactId>jackson2-api</artifactId>
+      <version>2.15.2-350.v0c2f3f8fc595</version>
+    </dependency>
+    <dependency>
+      <groupId>io.jenkins.plugins</groupId>
+      <artifactId>okhttp-api</artifactId>
+      <version>4.11.0-157.v6852a_a_fa_ec11</version>
+    </dependency>
```
Comment (jtnord) [inline pom.xml:259]: I do not depend on these and so they should not be needed.   What is the rationale for the the creep in dependencies?

perhaps you just need to pickup a newer bom version?

**Review by jtnord — CHANGES_REQUESTED** [review]

**Review by jtnord — APPROVED** [review]

**Pull request merged**